### ISSUE TITLE
Listing products doesn't paginate

### ISSUE THREAD

**Issue by petewall** [vmware-labs/marketplace-cli#9]
`mkpcli product list` only gets the first page of items. The http utilities should support fetching multiple pages of products and printing the whole list.

This has been low-priority, since the focus has been on functionality supporting editing single products, not discovery of products, but it would be nice to polish this area a bit.

**Comment by petewall** [comment]
It's worth noting that what little pagination already exists doesn't even work.

The PaginationObject that we generate makes a string like this:
`{"page": 1, "pageSize": 20}`, then we encode it onto the URL like this: `?pagination=%7B%22page%22%3A+1%2C+%22pageSize%22%3A+20%7D`

However, the Marketplace expects a parameter that looks like this:
`pagination={%22page%22:1,%22pageSize%22:20}`

So, only the `"`'s are url encoded.

This will have an impact on how we prepare requests, specifically the `MakeRequest` helper function. 


Note, if we hardcode the `RawQuery` to the expected string, it works:
```golang
	marketplaceURL := &url.URL{
		Scheme:   "https",
		Host:     viper.GetString("marketplace.host"),
		Path:     path,
		RawQuery: "pagination={%22page%22:1,%22pageSize%22:20}",
	}
```

So, to properly fix this, we want a clean way to pass the pagination object and have it encode in its special way.

**Comment by petewall** [comment]
Fixed in d85dab3
